src/include/parser/expression/function_expression.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+#include "parser/expression/abstract_expression.h"
+#include "type/type_id.h"
+
+namespace terrier {
+namespace parser {
+namespace expression {
+
+/**
+ * Represents a logical function expression.
+ */
+class FunctionExpression : public AbstractExpression {
+ public:
+  /**
+   * Instantiate a new function expression with the given name and children.
+   * @param func_name function name
+   * @param return_value_type function return value type
+   * @param children children arguments for the function
+   */
+  explicit FunctionExpression(std::string &&func_name, const type::TypeId return_value_type,
+                              std::vector<std::unique_ptr<AbstractExpression>> *children)
+      : AbstractExpression(ExpressionType::FUNCTION, return_value_type, std::move(*children)),
+        func_name_(std::move(func_name)) {}
+
+  AbstractExpression *Copy() const override { return new FunctionExpression(*this); }
+
+  /**
+   * Returns the name of the function.
+   * @return function name
+   */
+  const std::string &GetFuncName() const { return func_name_; }
+
+ protected:
+  // TODO(WAN): clang-tidy wants me to =default, how does that work?
```

Comment:
clang-tidy wants you to default because I believe you are doing the same thing a default copy-constructor does. Declaring something default might allow the compiler to reason some more about things.

Speaking of copy constructors. Why do we need one here and nowhere else?